src/renderer/features/fellowship-tasks/service.ts

```diff
@@ -0,0 +1,153 @@
+ * not close.
+ */
+function getControversyScore(referendum: OngoingReferendum, maximumAvailableVotingWeight: number): number {
+  const totalCast = referendum.tally.ayes + referendum.tally.nays;
+
+  if (totalCast == 0) {
+    return 0;
+  }
+
+  if (maximumAvailableVotingWeight == 0) {
+    return 0.0;
+  }
+
+  const fractionCast = totalCast / maximumAvailableVotingWeight;
+  const fractionAyes = referendum.tally.ayes / totalCast;
+
+  // closenessToSplit is 1 if fractionAyes = 0.5 and 0 if fractionAyes=0 or  fractionAyes=1
+  const closenessToSplit = 1.0 - Math.abs(0.5 - fractionAyes) * 2.0;
+
+  const controversy = fractionCast * closenessToSplit;
+
+  return Math.max(0.0, Math.min(1.0, controversy));
+}
+
+/**
+ * Returns how important the current user's vote might be, normalized to [0..1].
+ * 1 indicates that the user alone holds all (or nearly all) the voting power, 0
+ * indicates the user has no effective voting power relative to the total.
+ */
+function getUserImportanceScore(maximumAvailableVotingWeight: number, userVotingPower: number) {
+  if (maximumAvailableVotingWeight <= 0) {
+    return 0;
+  }
+
+  const importance = userVotingPower / maximumAvailableVotingWeight;
+  return Math.max(0.0, Math.min(1.0, importance));
+}
+
+/**
+ * Weights are adjustable and can be changed after feedback
+ */
+function getSortingScope(isUrgent: boolean, isControversial: boolean, isImportantVote: boolean) {
+  if (isUrgent && isImportantVote) {
+    return 4;
+  } else if (isUrgent && isControversial) {
+    return 3;
+  } else if (isImportantVote) {
+    return 2;
+  } else if (isControversial) {
+    return 1;
+  }
+
+  return 0;
+}
+
+/**
+ * Computes an overall importance (sorting_score) for a given referendum, along
+ * with a set of tags for UI to display The sorting_score is a simple linear
+ * combination of the three sub-scores
+ */
+function getReferendumImportance({
+  referendum,
+  maximumAvailableVotingWeight,
+  memberVotingWeight,
+  currentBlock,
+}: {
+  referendum: OngoingReferendum;
+  maximumAvailableVotingWeight: number;
+  memberVotingWeight: number;
+  currentBlock: BlockHeight;
+}): ReferendumImportance {
+  const urgencyScore = getUrgencyScore(referendum, currentBlock);
+  const controversyScore = getControversyScore(referendum, maximumAvailableVotingWeight);
+  const userImportanceScore = getUserImportanceScore(maximumAvailableVotingWeight, memberVotingWeight);
+
+  const isUrgent = urgencyScore > 0.3;
+  const isControversial = controversyScore > 0.5;
+  const isImportantVote = userImportanceScore > 0.5;
+
+  const tags: string[] = [];
+  const sortingScore = getSortingScope(isUrgent, isControversial, isImportantVote);
+
+  if (isUrgent) {
+    tags.push('urgent');
+  }
+  if (isControversial) {
+    tags.push('controversial');
+  }
+  if (isImportantVote) {
+    tags.push('importantVote');
+  }
+
+  return {
+    sortingScore,
+    tags,
+  };
```

Comment:
Personally I like inline objet with a couple properties, can skip it if you don't like
```suggestion
  return { sortingScore, tags };
```